File: ruby/raindrops/raindrops.rb
# Raindrops converts a number to raindrop-speak.
module Raindrops
  FACTORS = [
    [3, 'Pling'],
    [5, 'Plang'],
    [7, 'Plong'],
  ].freeze

  def self.convert(number)
    text = FACTORS.select { |f, _| (number % f).zero? }.map(&:last).join
    return number.to_s if text.empty?
    text
  end
end

module BookKeeping
  VERSION = 3
end

File: clojure/raindrops/src/raindrops.clj
(ns raindrops)

(defn- div? [n div] (zero? (mod n div)))

(defn convert [n]
  (if (some (partial div? n) [3 5 7])
    (str (if (div? n 3) "Pling" "") (if (div? n 5) "Plang" "") (if (div? n 7) "Plong" ""))
    (str n)))
